test/smtp.test.lua

```diff
@@ -1,13 +1,33 @@
 #!/usr/bin/env tarantool
 
 local tap = require('tap')
-local client = require('smtp').new()
 local test = tap.test("curl")
+local smtp = require('smtp')
 local fiber = require('fiber')
 local socket = require('socket')
 local os = require('os')
 local log = require('log')
 
+local client = smtp.new()
+
+local SEMVER_RE = '(%d+)%.(%d+)%.(%d+)'
+
+local function curl_version()
+    local curl_version_str = smtp.__CURL_VERSION
+    local major, minor, patch = string.match(curl_version_str, SEMVER_RE)
+    return tonumber(major), tonumber(minor), tonumber(patch)
+end
+
+-- Whether current curl version is greater or equal to the provided one.
+local function is_curl_version_ge(maj, min, patch)
+    local cur_maj, cur_min, cur_patch = curl_version()
+    return (cur_maj >= maj) and
+           (cur_min >= min) and
+           (cur_patch >= patch)
+end
```

Comment:
Is 3.2.1 >= 1.2.3? This function gives `false`.

There are some implementatoin of version comparison code you may be interested in: https://github.com/tarantool/tarantool/discussions/7006.